lib/EventParser.js

```diff
@@ -13,7 +13,7 @@ module.exports = function parseEvent(event) {
 
             const snsEvent = JSON.parse( eventRecord.Sns.Message );
             const snsEventRecord = snsEvent.Records && snsEvent.Records[0];
-            if ( snsEventRecord.eventSource === "aws:s3" && snsEventRecord.s3 ) {
+            if ( snsEventRecord && snsEventRecord.eventSource === "aws:s3" && snsEventRecord.s3 ) {
```

Comment:
Yes. It's when Lambda is installed and you add new handler on a bucket. Is such case AWS sends an [`TestEvent`](http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html) which was causing this code to crash. It's covered by one of the tests.